src/bugreporter/odemis_bugreporter.py

```diff
@@ -0,0 +1,542 @@
+        self.make_suggestion = True
+
+        # Load known users, if not available make tsv file
+        conf_path = os.path.join(os.path.expanduser(u"~"), '.config', 'odemis', 'bugreporter.tsv')
+        with open(conf_path, 'a+') as f:
+            reader = csv.reader(f, delimiter='\t')
+            self.known_users = OrderedDict(reader)
+
+    def store_user_info(self, name, email):
+        """
+        Store the user name and email in the config file, so it can be suggested the
+        next time the bugreporter is used.
+        :arg name: (String) user name
+        :arg email: (String) user email
+        """
+        # Add user to top of tsv file, truncate file if it contains too many users.
+        # Adding the user to the top of the list ensures that the suggestion is made based
+        # on the latest entry. Otherwise, if a typo occurred the first time the name was written,
+        # the faulty name will always be suggested.
+        if name in self.known_users.keys():
+            del self.known_users[name]
+        elif len(self.known_users.items()) >= MAX_USERS:
+            oldest_entry = self.known_users.keys()[-1]
+            del self.known_users[oldest_entry]
+        # It would be nicer not to create a new ordered dictionary, but to move the
+        # element internally. The python 3 version has such a function (move_to_end).
+        prev_items = self.known_users.items()
+        self.known_users = OrderedDict([(name, email)] + prev_items)
+        # Overwrite tsv file
+        conf_path = os.path.join(os.path.expanduser(u"~"), '.config', 'odemis', 'bugreporter.tsv')
+        with open(conf_path, 'w+') as f:
+            writer = csv.writer(f, delimiter='\t')
+            writer.writerows(self.known_users.items())
+
+    def on_name_key_down(self, evt):
+        """
+        If the pressed key is the backspace, delete suggestion and allow user to type new key.
+        """
+        if evt.GetKeyCode() == wx.WXK_BACK:
+            self.make_suggestion = False
+        else:
+            self.make_suggestion = True
+        evt.Skip()
+
+    def on_name_text(self, _):
+        """
+        Suggest a username from the configuration file.
+        """
+        if self.make_suggestion:
+            full_text = self.name_ctrl.GetValue()
+            if not full_text:
+                return
+
+            # Get typed text from input field (don't include suggestion)
+            if self.name_ctrl.GetSelection()[0] != self.name_ctrl.GetSelection()[1]:
+                typed = full_text[:len(full_text) - self.name_ctrl.GetSelection()[0] + 1].encode("utf-8")
+            else:
+                typed = full_text.encode("utf-8")
+
+            # Suggest name from configuration file and select suggested part
+            for name in self.known_users.keys():
+                if name.upper().startswith(typed.upper()):
+                    wx.CallAfter(self.name_ctrl.ChangeValue, name.decode("utf-8"))
```

Comment:
Note, this function is only called on a GUI event. So it's _always_ running in the GUI thread. So you don't need the wx.CallAfter(), as it's safe to immediately call the GUI functions.

Reply:
Actually, when I run it without CallAfter, it doesn't behave as intended. The idea is that the characters that are suggested are going to be highlighted with SetSelection. However, this doesn't work if I call it directly without the CallAfter. I don't know why this is, maybe you have an idea?

Reply:
The main different between with and without (in addition to the threading issue) is the call order. With the `CallAfter()` There might be extra thing happening in the text control after this function, and before the call to `ChangeValue()`.